apollo-federation/src/sources/connect/expand/mod.rs

```diff
@@ -799,36 +797,53 @@ mod helpers {
     fn extract_params_from_body(
         connector: &Connector,
     ) -> Result<HashSet<Parameter>, FederationError> {
-        let body_parameters = connector
-            .transport
-            .body
-            .as_ref()
-            .and_then(JSONSelection::extract_parameters)
-            .unwrap_or_default();
-
-        body_parameters
-            .iter()
-            .map(|StaticParameter { name, paths }| {
-                let mut parts = paths.iter();
-                let field = parts.next().ok_or(FederationError::internal(
-                    "expected parameter in JSONSelection to contain a field",
-                ))?;
-
-                match *name {
-                    "$args" => Ok(Parameter::Argument {
-                        argument: field,
-                        paths: parts.copied().collect(),
-                    }),
-                    "$this" => Ok(Parameter::Sibling {
-                        field,
-                        paths: parts.copied().collect(),
-                    }),
-                    other => Err(FederationError::internal(format!(
-                        "got unsupported parameter: {other}"
-                    ))),
-                }
-            })
-            .collect::<Result<HashSet<_>, _>>()
+        if let Some(body) = &connector.transport.body {
```

Comment:
You can do this to denest the rest of the function:

```rust
let Some(body) = &connector.transport.body else {
        return Ok(HashSet::default());
 };
```

Reply:
https://github.com/apollographql/router/pull/5762/commits/05f681dcf912f7916e8ba8aa445e2f0ea53733ee